pairtools/pairtools_dedup.py

```diff
@@ -1,248 +1,421 @@
+    "--c2",
+    type=int,
+    default=_pairsam_format.COL_C2,
+    help="Chrom 2 column; default {}".format(_pairsam_format.COL_C2),
+)
 @click.option(
-    "--p1", 
-    type=int, 
-    default=_pairsam_format.COL_P1,  
-    help='Position 1 column; default {}'.format(_pairsam_format.COL_P1))
+    "--p1",
+    type=int,
+    default=_pairsam_format.COL_P1,
+    help="Position 1 column; default {}".format(_pairsam_format.COL_P1),
+)
 @click.option(
-    "--p2", 
-    type=int, 
-    default=_pairsam_format.COL_P2,  
-    help='Position 2 column; default {}'.format(_pairsam_format.COL_P2))
+    "--p2",
+    type=int,
+    default=_pairsam_format.COL_P2,
+    help="Position 2 column; default {}".format(_pairsam_format.COL_P2),
+)
 @click.option(
-    "--s1", 
-    type=int, 
-    default=_pairsam_format.COL_S1,  
-    help='Strand 1 column; default {}'.format(_pairsam_format.COL_S1))
+    "--s1",
+    type=int,
+    default=_pairsam_format.COL_S1,
+    help="Strand 1 column; default {}".format(_pairsam_format.COL_S1),
+)
 @click.option(
-    "--s2", 
-    type=int, 
-    default=_pairsam_format.COL_S2,  
-    help='Strand 2 column; default {}'.format(_pairsam_format.COL_S2))
+    "--s2",
+    type=int,
+    default=_pairsam_format.COL_S2,
+    help="Strand 2 column; default {}".format(_pairsam_format.COL_S2),
+)
 @click.option(
-    "--unmapped-chrom", 
-    type=str, 
-    default=_pairsam_format.UNMAPPED_CHROM,  
-    help='Placeholder for a chromosome on an unmapped side; default {}'.format(_pairsam_format.UNMAPPED_CHROM))
+    "--unmapped-chrom",
+    type=str,
+    default=_pairsam_format.UNMAPPED_CHROM,
+    help="Placeholder for a chromosome on an unmapped side; default {}".format(
+        _pairsam_format.UNMAPPED_CHROM
+    ),
+)
 @click.option(
-    "--mark-dups", 
+    "--mark-dups",
     is_flag=True,
     help='If specified, duplicate pairs are marked as DD in "pair_type" and '
-         'as a duplicate in the sam entries.')
+    "as a duplicate in the sam entries.",
+)
 @click.option(
-    "--extra-col-pair", 
+    "--extra-col-pair",
     nargs=2,
-    #type=click.Tuple([str, str]),
+    # type=click.Tuple([str, str]),
     multiple=True,
-    help='Extra columns that also must match for two pairs to be marked as '
-    'duplicates. Can be either provided as 0-based column indices or as column '
+    help="Extra columns that also must match for two pairs to be marked as "
+    "duplicates. Can be either provided as 0-based column indices or as column "
     'names (requires the "#columns" header field). The option can be provided '
-    'multiple times if multiple column pairs must match. '
-    'Example: --extra-col-pair "phase1" "phase2"'
-    )
-
+    "multiple times if multiple column pairs must match. "
+    'Example: --extra-col-pair "phase1" "phase2"',
+)
+@click.option(
+    "--save-parent-id",
+    is_flag=True,
+    help="If specified, duplicate pairs are marked with the readID of the retained"
```

Comment:
Have you considered adding parent readID as a separate field?

Reply:
Not sure what you mean, that's what this flag will do...

Reply:
Yes, you're right, it's already a separate field. My apologies, I thought it substitutes the readID by the parental one. Might be good to specify what's the name of the generated field.

Reply:
Good idea, thanks!